--- a/ygoenv/ygoenv/ygopro/ygopro.h
+++ b/ygoenv/ygoenv/ygopro/ygopro.h
@@ -1824,11 +1824,11 @@
           "obs:h_actions_"_.Bind(
               Spec<uint8_t>({conf["n_history_actions"_], n_action_feats + 2})),
           "obs:mask_"_.Bind(Spec<uint8_t>({conf["max_cards"_] * 2, 14})),
-          "info:num_options"_.Bind(Spec<int>({}, {0, conf["max_options"_] - 1})),
-          "info:to_play"_.Bind(Spec<int>({}, {0, 1})),
-          "info:is_selfplay"_.Bind(Spec<int>({}, {0, 1})),
-          "info:win_reason"_.Bind(Spec<int>({}, {-1, 1})),
+          "info:num_options"_.Bind(Spec<int>({}, std::tuple<int, int>{0, conf["max_options"_] - 1})),
+          "info:to_play"_.Bind(Spec<int>({}, std::tuple<int, int>{0, 1})),
+          "info:is_selfplay"_.Bind(Spec<int>({}, std::tuple<int, int>{0, 1})),
+          "info:win_reason"_.Bind(Spec<int>({}, std::tuple<int, int>{-1, 1})),
           "info:step_time"_.Bind(Spec<double>({2})),
           "info:deck"_.Bind(Spec<int>({2})),
           // ADD THIS LINE
@@ -1837,7 +1837,7 @@
     static decltype(auto) ActionSpec(const Config &conf)
     {
       return MakeDict(
-          "action"_.Bind(Spec<int>({}, {0, conf["max_options"_] - 1})));
+          "action"_.Bind(Spec<int>({}, std::tuple<int, int>{0, conf["max_options"_] - 1})));
     }
   };
