--- a/repo/packages/y/ygopro-core/xmake.lua
+++ b/repo/packages/y/ygopro-core/xmake.lua
@@ -11,7 +11,7 @@ package("ygopro-core")
     on_install("linux", function (package)
         io.writefile("xmake.lua", [[
             add_rules("mode.debug", "mode.release")
-            add_requires("lua")
+            add_requires("lua", {system = true})
             target("ygopro-core")
                 set_kind("static")
                 add_files("*.cpp")
